site-packages/integralstor/manifest_status.py

```diff
@@ -160,53 +225,61 @@ def get_cpu_cores():
         return cores, None
 
 
-def get_mem_info():
+def get_mem_info_status(type='all'):
     """The total and the used memory information."""
     ret = {}
     try:
+        if type not in ['info', 'status', 'all']:
+            raise Exception('Invalid type passed.')
         mem_total = None
         mem_free = None
         with open('/proc/meminfo', 'r') as f:
             lines = f.readlines()
+            mem_total = {}
+            mem_free = {}
             for line in lines:
-                match = re.match(
-                    'MemTotal\s*:\s+([\S]+)\s+(\S+)$', line.strip())
-                if match:
-                    grps = match.groups()
-                    if grps:
-                        mem_total = {}
-                        mem_total['unit'] = grps[1]
-                        mem_total['value'] = int(grps[0])
-                        if mem_total['unit'] == 'kB':
-                            mem_total['unit'] = 'GB'
-                            mem_total['value'] = float(
-                                mem_total['value']) / 1024 / 1024
-                        elif mem_total['unit'] == 'mB':
-                            mem_total['unit'] = 'GB'
-                            mem_total['value'] = float(
-                                mem_total['value']) / 1024
-                match = re.match(
-                    'MemFree\s*:\s+([\S]+)\s+(\S+)$', line.strip())
-                if match:
-                    grps = match.groups()
-                    if grps:
-                        mem_free = {}
-                        mem_free['unit'] = grps[1]
-                        mem_free['value'] = int(grps[0])
-                        if mem_free['unit'] == 'kB':
-                            mem_free['unit'] = 'GB'
-                            mem_free['value'] = float(
-                                mem_free['value']) / 1024 / 1024
-                        elif mem_free['unit'] == 'mB':
-                            mem_free['unit'] = 'GB'
-                            mem_free['value'] = float(
-                                mem_total['value']) / 1024
-                if mem_total and mem_free:
-                    break
-        if not mem_total and not mem_free:
-            raise Exception('Error retrieving memory information')
-        ret['mem_total'] = mem_total
-        ret['mem_free'] = mem_free
+                if type in ['info', 'all']:
+                    match = re.match(
+                        'MemTotal\s*:\s+([\S]+)\s+(\S+)$', line.strip())
+                    if match:
+                        grps = match.groups()
+                        if grps:
+                            mem_total['unit'] = grps[1]
+                            mem_total['value'] = int(grps[0])
+                            if mem_total['unit'] == 'kB':
+                                mem_total['unit'] = 'GB'
+                                mem_total['value'] = int(float(
```

Comment:
I doubt if this will work for values less than a GB; some thing like 1001336 kB. `mem_total['value']` will be `0` in that case. Try this instead. 
```
mem_total['value'] = float(grps[0])
    if mem_total['unit'] == 'kB':
          mem_total['value'] = "%.2f" % ((mem_total['value'] / 1024) / 1024)
```

Rest of the following conditional checks have similar effects as well.

Reply:
True.. I did this because ZFS recommends 1GB per 1TB of disk space and no one nowadays will ever have less than a TB so.. And also, it really is a way of giving them/us a rough idea as the finer quantity does not really matter..

Reply:
Well, if free memory is less than a GB, showing it as '0.00 GB free' might be misleading.